### PR TITLE
Modularize try-except  ImportError

### PR DESCRIPTION
**Pull request by feipenghe**
This PR fixes https://github.com/asyml/forte/issues/747 

### Description of changes
* add a function that try-catch import error for a list of modules
* change all current hard-coded try-catch import errors to modularized ones.

### Possible influences of this PR.
* Future developers can use this method to report import error for third party packages in scripts.

### Test Conducted
* test import modules
* test raise errors when the module is not installed

### CODE REVIEW

File: forte/utils/utils.py [hunk 1]
```diff
@@ -303,3 +304,81 @@ def build_alignment(self, text: str, text_to_align: str):
...
+        parent_path, module_name, alias = module_info
+        if parent_path is None:
+            import_script = f"import {module_name}"
+        else:
+            import_script = f"from {parent_path} import {module_name} "
+        if alias is not None:
+            import_script += f" as {alias}"
+        if not top_level:
+            import_script += "# pylint: disable=import-outside-toplevel"
+        exec(import_script)
```
Comment (hunterhector) [inline forte/utils/utils.py:371]: I don't feel very confident about doing `exec`. Can we simply use importlib https://docs.python.org/3/library/importlib.html#importlib.import_module?

Are you trying to create the code here? I would rather not do this. And even if we need to do this, I don't think we need to include the `pylint` escpae here, I suppose `pylint` won't check the string you concatenated.

**Review by hunterhector — COMMENTED** [review]
The solution I suggested is actually much simpler, we just need to modularize the "error message", but don't need to do this.  You are working on a solution that tries to mimic the whole import behavior.

If this works then it might be fine, but I don't find it safe/reliable to use `exec` and manual code concat, can we simply use `importlib`?

If this solution has to use `exec` and then the issue is quite complicated, I would suggest still trying to modularize the error message only.

File: forte/utils/utils.py [hunk 2]
```diff
@@ -303,3 +304,45 @@ def build_alignment(self, text: str, text_to_align: str):
...
+    Args:
+        extra_module: module name should be installed by pip.
+        forte_module: forte module User should install by
+            ``pip install forte[`forte_module`]`` to install all
+            extra packages for using the forte module.
+        component_name: the forte component that needs the module.
+
+    """
+    forte_module_link_d = {
+        "data_aug": "https://github.com/asyml/forte/blame/master/README.md#L70",
```
Comment (hunterhector) [inline forte/utils/utils.py:328]: These links might be updated from time to time, like line number changes. Also, you should not use the `blame` version, even if this points to the correct line. 

A couple of solutions would be:
1. Consider adding link/anchor to that line
2. Or simply use a tagged version or a particular commit

You can point to the general area, without pointing to the specific line if that's not possible.

File: forte/processors/data_augment/algorithms/dictionary.py [hunk 3]
```diff
@@ -95,8 +95,9 @@ def __init__(self):
             )
         except ImportError as err:
             raise ImportError(
-                "nltk is not installed correctly."
-                "Please refer to documentation at [link](pip install forte[data_aug])"
+                create_import_error_msg(
+                    "nltk", "data_aug", "Data Augmentation(UDA)"
```
Comment (hunterhector) [inline forte/processors/data_augment/algorithms/dictionary.py:99]: This is a dictionary-based data augmentation, now UDA

File: forte/utils/utils.py [hunk 4]
```diff
@@ -303,3 +304,45 @@ def build_alignment(self, text: str, text_to_align: str):
...
+        "test": "https://github.com/asyml/forte/blame/master/README.md#L76",
+        "wikipedia": "https://github.com/asyml/forte/blame/master/README.md#L77",
+        "nlp": "https://github.com/asyml/forte/blame/master/README.md#L78",
+        "extractor": "https://github.com/asyml/forte/blame/master/README.md#L79",
+    }
+
+    error_msg = (
+        f" `{extra_module}` is not installed correctly."
+        + f" Consider install {extra_module}"
+        + f"via `pip install {extra_module}`."
```
Comment (hunterhector) [inline forte/utils/utils.py:343]: Sometimes the module may not be installed via `pip`, just tell the user to install this python module.

File: forte/utils/utils.py (hunk 2 above)
Comment (feipenghe) [reply forte/utils/utils.py:328]: To have anchors for these links, there should be section for each forte module like the screenshot below. Do you think it's acceptable? If so I will create a separate PR for it.
image

Comment (hunterhector) [reply forte/utils/utils.py:328]: i don't think we have to point to the exact line here, just point to this section for now until we know a better solution.

**Review by hunterhector — APPROVED** [review]

**Pull request merged**